adobe/react-spectrum
# Keep focus on last pressed column header through loading state

Closes https://github.com/adobe/react-spectrum/issues/4219

## ✅ Pull Request Checklist:


## 📝 Test Instructions:


Was unable to write a test in jest as I think jest doesn't mimic the browser focus behavior.

Created story in TableView > column header focus after loading. 
Steps:
1. Turn on NerdeFocus in dev console
2. Click the Height column to sort and observe focus goes to header
3. Table state will change to loading and focus should stay on column header
4. Table finishes loading and focus still on column header
5. Press enter on the (focused) column header again and note the tableview goes into a loading state
6. After the table finishes loading, observe the focus ring is around the empty tableview body

![Screenshot 2023-07-10 at 3 53 25 PM](https://github.com/adobe/react-spectrum/assets/3901877/d6a7ff36-7416-42aa-b57a-6f2e77620e65)


## 🧢 Your Project:


Adobe/Quarry
